addon/services/toast.ts

```diff
@@ -0,0 +1,138 @@
+import { later } from '@ember/runloop';
+import Service from '@ember/service';
+import { isEmpty } from '@ember/utils';
+
+const TIMEOUT = 5000;
+
+enum ToastType {
+  INFO = 'info',
+  WARNING = 'warning',
+  ERROR = 'error',
+  SUCCESS = 'success'
+}
+
+type ToastOptions = {
+  closeButton?: true;
+  tapToDismiss?: boolean;
+  onclick?: Function | null;
+};
+
+const DEFAULT_OPTIONS: ToastOptions = Object.freeze({
+  closeButton: true,
+  onclick: null,
+  tapToDismiss: true
+});
+
+function delegate(ctx: Toast, type: ToastType): Function {
```

Comment:
I think the type should actually be 
`function delegate(ctx: Toast, type: ToastType): (message: string, title: string, opts: ToastOptions = DEFAULT_OPTIONS) => void`

Reply:
we might be able to create a type from it. This would allow us to use the type inside your Toast class bellow (for success, error, warning, info "attributes")
I'd like to look into this with you 😇